operation/fhir-operation-bulkdata/src/main/java/org/linuxforhealth/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java

```diff
@@ -50,86 +40,54 @@ public static JobIdEncodingTransformer getInstance() {
-        SecretKeySpec secretKey = null;
-
-        if (encryptionKey != null && !encryptionKey.isEmpty()) {
-            try {
-                byte[] keyBytes = encryptionKey.getBytes("UTF-8");
-                keyBytes = Arrays.copyOf(MessageDigest.getInstance("SHA-1").digest(keyBytes), 16);
-                secretKey = new SecretKeySpec(keyBytes, "AES");
-            } catch (Exception e) {
-                logger.log(Level.WARNING, "Fail to generate encryption key from config!", e);
-            }
-        }
+    private Hashids createHashids() {
+        String salt = ConfigurationFactory.getInstance().getCoreBatchIdEncodingKey();
 
-        if (secretKey == null) {
-            logger.warning("Failed to get encryption key, JavaBatch Job ids will not be encrypted!");
+        if (salt != null && !salt.isEmpty()) {
+            return new Hashids(salt, MIN_ENCODED_LENGTH);
+        } else {
+            return null;
         }
-        return secretKey;
     }
 
     /**
      * encodes the job id
      *
-     * @implNote note a Cipher is used here, however it is not used to encrypt sensitive information rather encode the jobId.
-     *
      * @param jobId
      * @return
      */
-    public String encodeJobId(String jobId) {
+    public String encodeJobId(long jobId) {
         String tenantId = FHIRRequestContext.get().getTenantId();
-        SecretKeySpec key = KEY_MAP.computeIfAbsent(tenantId, k -> getJobIdEncryptionKey());
-        // Encrypt and UrlEncode the batch job id.
-        if (key == null) {
-            return jobId;
+        Hashids encoder = KEY_MAP.computeIfAbsent(tenantId, k -> createHashids());
+        if (encoder == null) {
+            return Long.toString(jobId);
         } else {
-            try {
-                // Use light weight encryption without salt to simplify both the encryption/decryption and also config.
-                Cipher cp = Cipher.getInstance("AES/ECB/PKCS5Padding");
-                cp.init(Cipher.ENCRYPT_MODE, key);
-
-                // Encrypt the job id, base64-encode it, and replace all `/` chars with the less problematic `_` char
-                String encodedJobId = Base64.getEncoder().withoutPadding().encodeToString(cp.doFinal(jobId.getBytes("UTF-8"))).replaceAll("/", "_");
-                // The encrypted job id is used in the polling content location url directly, so urlencode here.
-                return java.net.URLEncoder.encode(encodedJobId, StandardCharsets.UTF_8.name());
-            } catch (Exception e) {
-                return jobId;
-            }
+            return encoder.encode(jobId);
         }
     }
 
     /**
-     * decodes the job id.
-     *
-     * @implNote note a Cipher is used here, however it is not used to encrypt sensitive information rather encode the
-     *           jobId.
+     * decodes the job id
      *
      * @param encodedJobId
      * @return
+     * @throws IllegalArgumentException if the passed encodedJobId could not be decoded
```

Comment:
remove this old code?

Reply:
I chose to keep that part because we don't have a good way to know whether the jobId passed by the client is valid or not until we try decoding it.

Reply:
One thing we could (should?) do is look for invalid chars in the "encodedJobId" before we try decoding that...that would at least catch anything nefarious.
I can add that but I think we need to keep throwing if the encodedJobId doesn't decode as expected.

Reply:
actually I just checked the Hashids lib and its basically already doing this (right away when you call decode):
```
    String validChars = this.alphabet + this.guards + this.seps;
    for (int i = 0; i < hash.length(); i++) {
      if(validChars.indexOf(hash.charAt(i)) == -1) {
        return new long[0];
      }
    }
 ```
 
 so they chose to return an empty array.  in our case we're returning a String, but we could define a String of "-1" to be an error if we prefer.

Reply:
But is the KEY_MAP still required?

Reply:
oh, i thought i deleted this whole file... oops

Reply:
ok, its gone now.  good catch.